lib/hexdocs/search.ex

```diff
@@ -0,0 +1,80 @@
+defmodule Hexdocs.Search do
+  @moduledoc """
+  Module responsible for indexing the documentation.
+
+  Currently uses [Typesense Cloud.](https://cloud.typesense.org)
+  """
+
+  require Logger
+
+  @collection "hexdocs"
+
+  @spec process_results(String.t(), Version.t(), [map]) :: :ok
+  def index(package, version, search_items) do
+    full_package = "#{package}-#{version}"
+
+    ndjson =
+      Enum.map(search_items, fn item ->
+        json =
+          Map.take(item, ["type", "ref", "title", "doc"])
+          |> Map.put("package", full_package)
+          |> Jason.encode_to_iodata!()
+
+        [json, ?\n]
+      end)
+
+    url = url("collections/#{@collection}/documents/import?action=create")
+    headers = headers([{"content-type", "text/plain"}])
+
+    case :hackney.post(url, headers, ndjson) do
```

Comment:
Since it's possible to use real Typesense in tests, I'm not hiding it behind a behavior and make "real" HTTP requests here.

Reply:
I think this is fine but we should put it behind a flag that is excluded by default and add a docker-compose file that starts typesense locally.

Reply:
But if this is called from other code that is also tested then we need a "local" implementation essentially no-ops.

You can see this pattern for the Fastly code: https://github.com/hexpm/hexdocs/tree/d7a408dc94ea5389481d007baff03e385329638b/lib/hexdocs/cdn.

Reply:
> I think this is fine but we should put it behind a flag that is excluded by default 

This is done automatically right now if Typesense is not detected: https://github.com/hexpm/hexdocs/pull/44/files#diff-443233b9c679556d43885b3f3babee59b9c5c33da48887d591513e423d558f3bR7-R22

The warning also includes a Docker command to start Typesense. I'll try to make it appear only on `mix test --include typesense` instead of all the time.

> But if this is called from other code that is also tested

Right now no other code uses `search_data.js` files in tests, and that is one of the pre-requisets for Typesense indexing. I'll be refactoring this PR tomorrow, and I'll try putting Search behind a behaviour with Typesense being of the implementations.